python/TestHarness/TestHarness.py

```diff
@@ -370,6 +372,7 @@ def testOutputAndFinish(self, tester, retcode, output, start=0, end=0):
       (reason, output) = self.buildPBSBatch(output, tester)
     elif self.options.dry_run:
       reason = 'DRY_RUN'
+      test['status_message'] = reason
```

Comment:
Hmmm well, turns out --dry-run does not work with or with out this modification. Because this is a TestHarness specific option, we'll need to 'inject' this reason into a valid result response... since its not really a 'tester' object supplying the result response.